### PR TITLE
CB-17712 Check the backup before the resize

### PR DESCRIPTION
**Pull request by RokolyA94**
See detailed description in the commit message.

### CODE REVIEW

File: datalake/src/main/resources/duties/7.2.16/aws/cdp_data_lake_medium_duty_with_profiler/medium_duty_ha.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
             "size": 250,
             "type": "standard"
           }
-        ]
+        ]CB-17712
```
Comment (sxxgrc) [inline datalake/src/main/resources/duties/7.2.16/aws/cdp_data_lake_medium_duty_with_profiler/medium_duty_ha.json:55]: What is this for?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/MockSdxUpgradeTests.java [hunk 2]
```diff
@@ -79,6 +79,10 @@ public void testSdxUpgradeWhenNoAttachedDisksButEmbeddedDBShouldHaveBeenOnSelfCo
         String cluster = "cmcluster";
         String imageSettings = "imageSettingsUpgrade";
         String networkKey = "someOtherNetwork";
+        testContext
```
Comment (sxxgrc) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/MockSdxUpgradeTests.java:82]: I know you added this last time but do you mind explaining why these are needed? We've never needed something like this for mock tests before right? I don't see why your changes would require this for the upgrade tests.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java [hunk 3]
```diff
@@ -590,6 +598,18 @@ public boolean shouldSdxRestoreBePerformed(SdxCluster cluster, boolean entitleme
         return true;
     }
 
+    public void checkExistingBackup(SdxCluster newSdxCluster, String userId) {
+        datalakeDRProto.DatalakeBackupInfo lastSuccessfulBackup = datalakeDrClient
```
Comment (sxxgrc) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java:602]: Okay so what we can do here is add this check: `if (!datalakeDrConfig.isConfigured())` and if its `true` then we know that the environment we are running in does not have the DR service enabled, which should only occur for local environments set up that way and, currently, the E2E testing environments.

With this we can just skip the check here and return immediately. You would need to make proper modifications to your unit tests.

File: datalake/src/main/resources/duties/7.2.16/aws/cdp_data_lake_medium_duty_with_profiler/medium_duty_ha.json (hunk 1 above)
Comment (RokolyA94) [reply datalake/src/main/resources/duties/7.2.16/aws/cdp_data_lake_medium_duty_with_profiler/medium_duty_ha.json]: fixed.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java (hunk 3 above)
Comment (RokolyA94) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java:605]: done. Thank you!

**Review by kkalvagadda1 — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java [hunk 4]
```diff
@@ -118,6 +118,9 @@ public FlowIdentifier triggerSdxResize(Long sdxClusterId, SdxCluster newSdxClust
         boolean performRestore = sdxBackupRestoreService.shouldSdxRestoreBePerformed(
                 newSdxCluster, entitlementService.isDatalakeBackupOnResizeEnabled(ThreadBasedUserCrnProvider.getAccountId())
         );
+        if (!performBackup) {
+            sdxBackupRestoreService.checkExistingBackup(newSdxCluster, userId);
```
Comment (keyki) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java:122]: Isn't this call going to fail? We don't have the DR service in the QE environment. We should have a check here whether the service is configured or not.

Comment (RokolyA94) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java:122]: The `checkExistingBackup` method contains the following condition:
```java
        if (!datalakeDrConfig.isConfigured()) {
            return;
        }
```

Therefore the QE env will not failed on this.

Comment (keyki) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java:122]: Ah, missed that, thank you.

**Pull request merged**